### PR TITLE
feat: add openedge support

### PR DESCRIPTION
**Pull request by gquerret**
Language Server support for [OpenEdge](https://www.progress.com/openedge)
Based on the VSCode plugin, everything should be ok in this first version.

### CODE REVIEW

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          'java',
+          '--add-opens=java.base/java.lang=ALL-UNNAMED',
+          '--add-opens=java.base/java.math=ALL-UNNAMED',
+          '--add-opens=java.base/java.util=ALL-UNNAMED',
+          '--add-opens=java.base/java.util.concurrent=ALL-UNNAMED',
+          '--add-opens=java.base/java.net=ALL-UNNAMED',
+          '--add-opens=java.base/java.text=ALL-UNNAMED',
+        }
+        if config.debug then
+          table.insert(config.cmd, '-Dorg.slf4j.simpleLogger.defaultLogLevel=DEBUG')
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:19]:
```suggestion
          config.cmd[#config.cmd + 1] =  '-Dorg.slf4j.simpleLogger.defaultLogLevel=DEBUG'
```

Comment (glepnir) [reply lua/lspconfig/server_configurations/openedge_ls.lua:19]: a tip . in luajit this is faster than invoke `table.insert`

Comment (glepnir) [reply lua/lspconfig/server_configurations/openedge_ls.lua:19]: can you change it ?

**Comment by gquerret** [comment]
I'll do the modification beginning of next week.

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          '--add-opens=java.base/java.math=ALL-UNNAMED',
+          '--add-opens=java.base/java.util=ALL-UNNAMED',
+          '--add-opens=java.base/java.util.concurrent=ALL-UNNAMED',
+          '--add-opens=java.base/java.net=ALL-UNNAMED',
+          '--add-opens=java.base/java.text=ALL-UNNAMED',
+        }
+        if config.debug then
+          config.cmd[#config.cmd + 1] =  '-Dorg.slf4j.simpleLogger.defaultLogLevel=DEBUG'
+        end
+        table.insert(config.cmd, '-jar')
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:21]:
```suggestion
        config.cmd[#config.cmd + 1] = '-jar'
```

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        table.insert(config.cmd, config.oe_jar_path)
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:22]:
```suggestion
        config.cmd[#config.cmd + 1] = config.oe_jar_path
```

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        if config.dlc then
+          table.insert(config.cmd, '--dlc')
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:24]:
```suggestion
          config.cmd[#config.cmd + 1] = '--dlc'
```

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          table.insert(config.cmd, config.dlc)
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:25]:
```suggestion
          config.cmd[#config.cmd + 1] = config.dlc
```

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        end
+        if config.trace then
+          table.insert(config.cmd, '--trace')
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:28]:
```suggestion
          config.cmd[#config.cmd + 1] = '--trace'
```

File: lua/lspconfig/server_configurations/openedge_ls.lua [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+          'java',
+          '--add-opens=java.base/java.lang=ALL-UNNAMED',
+          '--add-opens=java.base/java.math=ALL-UNNAMED',
+          '--add-opens=java.base/java.util=ALL-UNNAMED',
+          '--add-opens=java.base/java.util.concurrent=ALL-UNNAMED',
+          '--add-opens=java.base/java.net=ALL-UNNAMED',
+          '--add-opens=java.base/java.text=ALL-UNNAMED',
+        }
+        if config.debug then
+          config.cmd[#config.cmd + 1] =  '-Dorg.slf4j.simpleLogger.defaultLogLevel=DEBUG'
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/openedge_ls.lua:19]:
```suggestion
          config.cmd[#config.cmd + 1] = '-Dorg.slf4j.simpleLogger.defaultLogLevel=DEBUG'
```

**Pull request merged**